pallene/checker.lua

```diff
@@ -53,25 +54,34 @@ function checker.check(prog_ast)
     end
 end
 
-local function checker_error(loc, fmt, ...)
-    local err_msg = location.format_error(loc, fmt, ...)
-    coroutine.yield(err_msg)
+-- Usually if an error is produced with `assert()` or `error()` then it is
+-- a compiler bug. User-facing errors such as syntax errors and
+-- type checking errors are reported in a different way. The actual
+-- method employed is kind of tricky. Since Lua does not have a clean
+-- try-catch functionality we use coroutines to do that job.
+-- You can see this in the `checker.check()` function but you do
+-- not need to know how it works. You just need to know that calling
+-- `scope_error()` or `type_error()` will exit the type checking routine
+-- and report a Pallene compilation error.
+local function checker_error(error_location, format, ...)
+    local error_message = location.format_error(error_location, format, ...)
+    coroutine.yield(error_message)
 end
 
-local function scope_error(loc, fmt, ...)
-    return checker_error(loc, ("scope error: " .. fmt), ...)
+local function scope_error(error_location, format, ...)
+    return checker_error(error_location, ("scope error: " .. format), ...)
```

Comment:
We are already using `loc` and `fmt` in other places in the code. I think it would be more consistent to keep the old names.

Reply:
Alright. I'll revert the changes I made.